qiskit/circuit/quantumcircuit.py

```diff
@@ -3386,6 +3386,91 @@ def depth(
 
         return max(op_stack)
 
+    def depth_2q(self, include_directives: bool = False) -> int:
+        """Returns the depth in terms of quantum gates with at least 2-qubits,
```

Comment:
`num_nq_gates` exactly checks the number of qubits because I thought users might be interested in that quantity in general. Also, `num_nonlocal_gates` determines the number of gates with at least two qubits. I extended the arguments of `num_nonlocal_gates` to include an integer `n` that specifies the minimum number of qubits in a gate to still be counted towards the quantity returned by `num_nonlocal_gates`. The default behavior of `num_nonlocal_gates` stays the same, so I hope our stability policy is not violated...

Reply:
Agree the name is a bit confusing an inconsistent with the others. Consider changing to something like `depth_multi_qubit`

Reply:
I was also thinking about whether `depth_nq` should work on gates with exactly `n` qubits or for `>= n` qubits. I decided for the latter because I thought the reason an user might want `depth_nq` to exclude certain gates is that they are comparatively cheap. I think it is safe to assume that if an user regards a gate with `n` qubits to be relatively expensive, they would also consider a gate with `n+i` qubits to be expensive (if there are no other properties to decide on). I also was not able to see a situation where an user might only be interested in the two-qubit gate depth while ignoring all other gates in the presence of e.g. three-qubit gates.